storage/remote/write_handler.go

```diff
@@ -92,6 +104,45 @@ func NewWriteHandler(logger *slog.Logger, reg prometheus.Registerer, appendable
 	return h
 }
 
+// validateV1LabelValueLength checks label values in v1 protobuf format before ToLabels() is called.
+// This prevents panics during deserialization of oversized labels.
+func validateV1LabelValueLength(labels []prompb.Label) (labelName string, valueLength int, exceeded bool) {
```

Comment:
If we do this, could we inline this? It's a shallow function, could be more readable inlined (signature is more complex than  than the body). It's fine to keep v2 validation in a function (it's more involved perhaps).